### PR TITLE
Update DI docs for v8

### PR DESCRIPTION
**Pull request by timbussmann**

### CODE REVIEW

File: Snippets/Core/Core_6/Hosting/Hosting.cs [hunk 1]
```diff
@@ -57,21 +57,5 @@ public static void SetInstance(IEndpointInstance endpoint)
             }
         }
         #endregion
-
-        async Task InjectEndpoint()
```
Comment (mauroservienti) [inline Snippets/Core/Core_6/Hosting/Hosting.cs:61]: @timbussmann why this change to Core 6 snippets?

File: Snippets/Core/Core_7/Hosting/Hosting.cs [hunk 2]
```diff
@@ -57,21 +57,5 @@ public static void SetInstance(IEndpointInstance endpoint)
             }
         }
         #endregion
-
-        async Task InjectEndpoint()
```
Comment (mauroservienti) [inline Snippets/Core/Core_7/Hosting/Hosting.cs:61]: @timbussmann why this change to Core 7 snippets?

File: Snippets/Core/Core_6/Hosting/Hosting.cs (hunk 1 above)
Comment (timbussmann) [reply Snippets/Core/Core_6/Hosting/Hosting.cs:61]: @mauroservienti I removed the whole snippet and related doco section as I didn't think it was very clear or useful and it also requires autofac as a dependency just for that snippet. Thoughts?
